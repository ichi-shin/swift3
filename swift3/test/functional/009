#!/bin/bash

# GET Bucket lifecycle

. ./common

S3USER=tester

_gen_lifecycle_conf()
{
    local n=$1

    echo '<LifecycleConfiguration>'

    for i in `seq $n`; do
	echo '<Rule>'
	echo "<ID>rule_${i}</ID>"
	echo "<Prefix>prefix${i}_</Prefix>"
	echo "<Status>Enabled</Status>"
	echo "<Expiration>"
	if [ $(($i % 2)) == 0 ]; then
	    echo "<Date>2014-12-31T00:00:00.000Z</Date>"
	else
	    echo "<Days>1</Days>"
	fi
	echo "</Expiration>"
	echo '</Rule>'
    done

    echo '</LifecycleConfiguration>'
}

_s3_put /bucket

for i in 1 2 100; do
    _gen_lifecycle_conf $i > $tmp.lifecycle.$i.xml
    MD5=$(_md5 $tmp.lifecycle.$i.xml) _s3_put /bucket?lifecycle -T $tmp.lifecycle.$i.xml
    _s3_get /bucket?lifecycle | _filter_curl xml
done
