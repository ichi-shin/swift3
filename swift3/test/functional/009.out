QA output created by 009
> s3curl --id tester -- -X PUT http://SWIFT_HOST/bucket... 200
> s3curl --id tester --contentMd5 3tpXw46ma2ltzOYvkwxK5A== -- -X PUT -T /TMP.lifecycle.1.xml http://SWIFT_HOST/bucket?lifecycle... 200
> s3curl --id tester -- -X GET http://SWIFT_HOST/bucket?lifecycle... 200
<?xml version='1.0' encoding='UTF-8'?>
<LifecycleConfiguration xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Rule>
    <ID>rule_1</ID>
    <Prefix>prefix1_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
</LifecycleConfiguration>
> s3curl --id tester --contentMd5 LSMf5zyuVtf2Qatlqy1oIw== -- -X PUT -T /TMP.lifecycle.2.xml http://SWIFT_HOST/bucket?lifecycle... 200
> s3curl --id tester -- -X GET http://SWIFT_HOST/bucket?lifecycle... 200
<?xml version='1.0' encoding='UTF-8'?>
<LifecycleConfiguration xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Rule>
    <ID>rule_1</ID>
    <Prefix>prefix1_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_2</ID>
    <Prefix>prefix2_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
</LifecycleConfiguration>
> s3curl --id tester --contentMd5 uQNmp+vcNfiGMaYUBSekZQ== -- -X PUT -T /TMP.lifecycle.100.xml http://SWIFT_HOST/bucket?lifecycle... 200
> s3curl --id tester -- -X GET http://SWIFT_HOST/bucket?lifecycle... 200
<?xml version='1.0' encoding='UTF-8'?>
<LifecycleConfiguration xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Rule>
    <ID>rule_1</ID>
    <Prefix>prefix1_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_2</ID>
    <Prefix>prefix2_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_3</ID>
    <Prefix>prefix3_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_4</ID>
    <Prefix>prefix4_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_5</ID>
    <Prefix>prefix5_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_6</ID>
    <Prefix>prefix6_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_7</ID>
    <Prefix>prefix7_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_8</ID>
    <Prefix>prefix8_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_9</ID>
    <Prefix>prefix9_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_10</ID>
    <Prefix>prefix10_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_11</ID>
    <Prefix>prefix11_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_12</ID>
    <Prefix>prefix12_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_13</ID>
    <Prefix>prefix13_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_14</ID>
    <Prefix>prefix14_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_15</ID>
    <Prefix>prefix15_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_16</ID>
    <Prefix>prefix16_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_17</ID>
    <Prefix>prefix17_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_18</ID>
    <Prefix>prefix18_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_19</ID>
    <Prefix>prefix19_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_20</ID>
    <Prefix>prefix20_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_21</ID>
    <Prefix>prefix21_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_22</ID>
    <Prefix>prefix22_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_23</ID>
    <Prefix>prefix23_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_24</ID>
    <Prefix>prefix24_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_25</ID>
    <Prefix>prefix25_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_26</ID>
    <Prefix>prefix26_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_27</ID>
    <Prefix>prefix27_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_28</ID>
    <Prefix>prefix28_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_29</ID>
    <Prefix>prefix29_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_30</ID>
    <Prefix>prefix30_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_31</ID>
    <Prefix>prefix31_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_32</ID>
    <Prefix>prefix32_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_33</ID>
    <Prefix>prefix33_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_34</ID>
    <Prefix>prefix34_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_35</ID>
    <Prefix>prefix35_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_36</ID>
    <Prefix>prefix36_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_37</ID>
    <Prefix>prefix37_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_38</ID>
    <Prefix>prefix38_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_39</ID>
    <Prefix>prefix39_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_40</ID>
    <Prefix>prefix40_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_41</ID>
    <Prefix>prefix41_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_42</ID>
    <Prefix>prefix42_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_43</ID>
    <Prefix>prefix43_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_44</ID>
    <Prefix>prefix44_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_45</ID>
    <Prefix>prefix45_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_46</ID>
    <Prefix>prefix46_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_47</ID>
    <Prefix>prefix47_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_48</ID>
    <Prefix>prefix48_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_49</ID>
    <Prefix>prefix49_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_50</ID>
    <Prefix>prefix50_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_51</ID>
    <Prefix>prefix51_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_52</ID>
    <Prefix>prefix52_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_53</ID>
    <Prefix>prefix53_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_54</ID>
    <Prefix>prefix54_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_55</ID>
    <Prefix>prefix55_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_56</ID>
    <Prefix>prefix56_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_57</ID>
    <Prefix>prefix57_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_58</ID>
    <Prefix>prefix58_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_59</ID>
    <Prefix>prefix59_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_60</ID>
    <Prefix>prefix60_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_61</ID>
    <Prefix>prefix61_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_62</ID>
    <Prefix>prefix62_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_63</ID>
    <Prefix>prefix63_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_64</ID>
    <Prefix>prefix64_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_65</ID>
    <Prefix>prefix65_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_66</ID>
    <Prefix>prefix66_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_67</ID>
    <Prefix>prefix67_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_68</ID>
    <Prefix>prefix68_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_69</ID>
    <Prefix>prefix69_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_70</ID>
    <Prefix>prefix70_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_71</ID>
    <Prefix>prefix71_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_72</ID>
    <Prefix>prefix72_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_73</ID>
    <Prefix>prefix73_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_74</ID>
    <Prefix>prefix74_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_75</ID>
    <Prefix>prefix75_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_76</ID>
    <Prefix>prefix76_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_77</ID>
    <Prefix>prefix77_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_78</ID>
    <Prefix>prefix78_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_79</ID>
    <Prefix>prefix79_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_80</ID>
    <Prefix>prefix80_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_81</ID>
    <Prefix>prefix81_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_82</ID>
    <Prefix>prefix82_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_83</ID>
    <Prefix>prefix83_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_84</ID>
    <Prefix>prefix84_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_85</ID>
    <Prefix>prefix85_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_86</ID>
    <Prefix>prefix86_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_87</ID>
    <Prefix>prefix87_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_88</ID>
    <Prefix>prefix88_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_89</ID>
    <Prefix>prefix89_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_90</ID>
    <Prefix>prefix90_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_91</ID>
    <Prefix>prefix91_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_92</ID>
    <Prefix>prefix92_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_93</ID>
    <Prefix>prefix93_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_94</ID>
    <Prefix>prefix94_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_95</ID>
    <Prefix>prefix95_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_96</ID>
    <Prefix>prefix96_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_97</ID>
    <Prefix>prefix97_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_98</ID>
    <Prefix>prefix98_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_99</ID>
    <Prefix>prefix99_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Days>1</Days>
    </Expiration>
  </Rule>
  <Rule>
    <ID>rule_100</ID>
    <Prefix>prefix100_</Prefix>
    <Status>Enabled</Status>
    <Expiration>
      <Date>DATE</Date>
    </Expiration>
  </Rule>
</LifecycleConfiguration>
